src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -236,14 +236,14 @@ public async Task<HttpResponseMessage> SendAsync(CancellationToken cancellationT
             catch (QuicStreamAbortedException ex)
             {
                 // Our stream was reset.
-
+                _requestBodyCancellationSource.Cancel();
```

Comment:
That's in order to cancel the `CancellationToken` passed to the `HttpContent`, Since the request is being torn down, there is no need for the (async) writes to the stream to continue.

Reply:
I have rather opposite question. Why is the `_requestBodyCancellationSource` cancelled only selectively in certain cases here?

What will happen with `sendContentTask` in duplex scenario in other exceptional cases?

Reply:
Yes, that was also my point. I am not yet convinced it is needed, but if it is, it should happen on all exception cases.

Reply:
As for whether the cancellation here is needed or not. I believe the cancellation of this token in general (i.e. the issue we fix here) is important for streaming scenarios, where the writes might not happen regularly so the abort of the connection etc might not be observed in time. The question is whether there is a scenario where such "unregular" streaming occurs before SendAsync finishes.

Reply:
Cancellation moved to the finally block